src/Control/HTTPApplication.php

```diff
@@ -141,4 +152,82 @@ public function execute(HTTPRequest $request, callable $callback, $flush = false
+     */
+    private function warnAboutDeprecatedSetups()
+    {
+        // TypeCreator is a class unique to GraphQL v3 - we use it in other areas to detect
+        // which version is being used.
+        if (class_exists(TypeCreator::class)) {
+            Deprecation::notice(
+                '4.13.0',
+                'silverstripe/graphql 3.x is deprecated. Upgrade to 4.x instead.'
+                . ' See https://docs.silverstripe.org/en/4/upgrading/upgrading_to_graphql_4/',
+                Deprecation::SCOPE_GLOBAL
+            );
+        }
+
+        // The alternate_public_dir config property is deprecated, but because it's
+        // always fetched it'll throw a deprecation warning whether you've set it or not.
+        // There are also multiple mechanisms which can result in this bad configuration.
+        if (PUBLIC_DIR !== 'public' || Director::publicDir() !== PUBLIC_DIR) {
+            Deprecation::notice(
+                '4.13.0',
+                'Use of a public webroot other than "public" is deprecated.'
+                . ' See https://docs.silverstripe.org/en/4/changelogs/4.1.0#public-folder/',
+                Deprecation::SCOPE_GLOBAL
+            );
+        }
+
+        // This change of defaults has no other deprecation notice being emitted currently.
+        $project = new Module(BASE_PATH, BASE_PATH);
+        if (RESOURCES_DIR !== '_resources' && $project->getResourcesDir() !== RESOURCES_DIR) {
+            Deprecation::notice(
+                '4.13.0',
+                'The RESOURCES_DIR constant will change to "_resources" by default.'
+                . ' See https://docs.silverstripe.org/en/5/changelogs/5.0.0/#api-general',
+                Deprecation::SCOPE_GLOBAL
+            );
+        }
+
+        // File resolution strategy defaults changed in CMS 4.4 - people may not have changed their
+        // config since then, so alert them their legacy strategy won't work in the future.
+        $publicResolutionStrat = Injector::inst()->get(FileResolutionStrategy::class . '.public');
+        $protectedResolutionStrat = Injector::inst()->get(FileResolutionStrategy::class . '.protected');
+        if (method_exists($publicResolutionStrat, 'getResolutionFileIDHelpers')) {
+            $hasDeprecatedStrategy = false;
+            foreach ($publicResolutionStrat->getResolutionFileIDHelpers() as $helper) {
+                if ($helper instanceof LegacyFileIDHelper) {
+                    $hasDeprecatedStrategy = true;
+                    break;
+                }
+            }
+
+            if (!$hasDeprecatedStrategy && method_exists($protectedResolutionStrat, 'getResolutionFileIDHelpers')) {
+                foreach ($protectedResolutionStrat->getResolutionFileIDHelpers() as $helper) {
+                    if ($helper instanceof NaturalFileIDHelper) {
+                        $hasDeprecatedStrategy = true;
+                        break;
+                    }
+                }
+            }
```

Comment:
AAAHHHHH okay so there's some further thinking we need here....
You know that lovely [config in installer](https://github.com/silverstripe/silverstripe-installer/blob/4/app/_config/assets.yml) that overrides the [asset config](https://github.com/silverstripe/silverstripe-assets/blob/1/_config/asset.yml#L25-L43)?
Config doesn't override - it merges. So ever since this was introduced people have actually had the following config:
```yml
Silverstripe\Core\Injector\Injector:
  SilverStripe\Assets\FilenameParsing\FileResolutionStrategy.public:
    class: SilverStripe\Assets\FilenameParsing\FileIDHelperResolutionStrategy
    properties:
      ResolutionFileIDHelpers:
        - '%$SilverStripe\Assets\FilenameParsing\HashFileIDHelper'
        - '%$SilverStripe\Assets\FilenameParsing\NaturalFileIDHelper'
        - '%$SilverStripe\Assets\FilenameParsing\LegacyFileIDHelper'
        - '%$SilverStripe\Assets\FilenameParsing\HashFileIDHelper'
        - '%$SilverStripe\Assets\FilenameParsing\NaturalFileIDHelper'
      DefaultFileIDHelper: '%$SilverStripe\Assets\FilenameParsing\NaturalFileIDHelper'
      VersionedStage: Live
  SilverStripe\Assets\FilenameParsing\FileResolutionStrategy.protected:
    class: SilverStripe\Assets\FilenameParsing\FileIDHelperResolutionStrategy
    properties:
      DefaultFileIDHelper: '%$SilverStripe\Assets\FilenameParsing\HashFileIDHelper'
      ResolutionFileIDHelpers:
        - '%$SilverStripe\Assets\FilenameParsing\HashFileIDHelper'
        - '%$SilverStripe\Assets\FilenameParsing\NaturalFileIDHelper'
        - '%$SilverStripe\Assets\FilenameParsing\HashFileIDHelper'
      VersionedStage: Stage
```

So _everyone_ is running the legacy (and then some) strategy.

Reply:
I've raised https://github.com/silverstripe/silverstripe-assets/pull/543 to emit this notice when a file is found in a deprecated file structure instead.